### PR TITLE
Valgrind fixes in modules

### PR DESCRIPTION
**Pull request by brianmoose**
Refs #6003

### CODE REVIEW

File: modules/phase_field/src/materials/DerivativeParsedMaterialHelper.C [hunk 1]
```diff
@@ -154,6 +157,9 @@ DerivativeParsedMaterialHelper::assembleDerivatives()
     // remove the 'current' element from the queue
     queue.pop_front();
   }
+  // cleanup zero derivatives
+  for (std::vector<ADFunction*>::iterator it = adfunctions.begin(); it != adfunctions.end(); ++it)
+    delete *it;
```
Comment (dschwen) [inline modules/phase_field/src/materials/DerivativeParsedMaterialHelper.C]: :+1:

File: modules/phase_field/src/auxkernels/TestEBSDAux.C [hunk 2]
```diff
@@ -24,6 +24,11 @@ TestEBSDAux::TestEBSDAux(const InputParameters & parameters) :
 {
 }
 
+TestEBSDAux::~TestEBSDAux()
+{
+  delete _val;
```
Comment (jwpeterson) [inline modules/phase_field/src/auxkernels/TestEBSDAux.C]: This is pretty annoying.  I don't really like the `EBSDReader::getPointDataAccessFunctor` interface, it just allocates memory and hands it off to the user, which is not really like other things in MOOSE and very prone to creating memory leaks, like we see here...

Comment (jwpeterson) [inline modules/phase_field/src/auxkernels/TestEBSDAux.C]: I'm merging this as-is but I want to change that interface to return a `MooseSharedPointer` instead.

Comment (dschwen) [inline modules/phase_field/src/auxkernels/TestEBSDAux.C]: Hmm, I don't remember why I new anything there. On the phone now but I'll look later if we can change that to references.

Comment (dschwen) [inline modules/phase_field/src/auxkernels/TestEBSDAux.C]: Or shared pointer, that'd be fine too.

Comment (jwpeterson) [inline modules/phase_field/src/auxkernels/TestEBSDAux.C]: Yes, it's a "build" type method that probably does need to allocate memory in order to work correctly, we just need to manage its lifetime better.

File: modules/phase_field/src/materials/DerivativeParsedMaterialHelper.C (hunk 1 above)
Comment (dschwen) [inline modules/phase_field/src/materials/DerivativeParsedMaterialHelper.C]: On second thought I think I'd rather change all of this to use `MooseSharedPointers` internally. That makes the code shorter by having automatic cleanup.

**Pull request merged**